src/main/java/org/elasticsearch/node/internal/InternalSettingsPreparer.java

```diff
@@ -38,10 +40,35 @@
  */
 public class InternalSettingsPreparer {
 
+    public static final String SECRET_PROMPT_VALUE = "__prompt:secret__";
```

Comment:
so, I've been thinking about it. We already have property place holders in the form `${...}`. I think it'd be more consistent if we use the same form except with special syntax inside... something like: `${prompt::text}`. @clintongormley wdyt?